File: hw/aymacana/u2/hw19Inheritance/FarmSystemPY/Pig.py
from datetime import datetime
from FarmAnimal import FarmAnimal
from Cage import Cage

class Pig(FarmAnimal):
    def __init__(self, id: int, breed: str, born_on: datetime, gender: str, 
                 is_able_to_reproduce: bool, weight: float, cage: Cage):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    
    def __str__(self) -> str:
        return f"Pig{{{super().__str__()}}}"

File: hw/pastor/u2/hw19 Inheritance/Javascript/ec.edu.espe.farmsystem.model/pig.js
const FarmAnimal = require('./farmAnimal');

class Pig extends FarmAnimal {
    constructor(isReadyForSlaughter, id, breed, bornOn, gender, isAbleToReproduce, weight, cage, location) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage, location);
        this.isReadyForSlaughter = isReadyForSlaughter;
    }

    toString() {
        const readyStr = this.isReadyForSlaughter ? "YES" : "No";
        return super.toString() + `
 > Ready to Kill : ${readyStr}
========================================`;
    }
}

module.exports = Pig;